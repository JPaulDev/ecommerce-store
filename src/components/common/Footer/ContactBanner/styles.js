import styled from 'styled-components';

export const Container = styled.div`
  background-color: var(--color-blue-3);
  display: flex;
  justify-content: center;
  padding: 15px 0;

  & :last-child {
    border: 0;
  }
`;

export const Link = styled.a`
  align-items: center;
  border-right: 2px solid white;
  color: white;
  column-gap: 10px;
  display: flex;
  font-family: 'Uni Sans', sans-serif;
  font-size: var(--font-size-20);
  font-weight: var(--font-weight-semibold);
  padding: 0 25px;
  text-decoration: none;

  &:hover {
    color: var(--color-grey-11);
  }
`;
