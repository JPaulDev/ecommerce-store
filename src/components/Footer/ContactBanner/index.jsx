import MaterialSymbol from '../../MaterialSymbol';
import * as Styled from './styles';

export default function ContactBanner() {
  return (
    <Styled.Container>
      <Styled.Link href="">
        <MaterialSymbol iconSize="28px">phone_iphone</MaterialSymbol>
        01782-444455
      </Styled.Link>
      <Styled.Link href="">
        <MaterialSymbol iconSize="28px">support_agent</MaterialSymbol>
        Live Chat
      </Styled.Link>
    </Styled.Container>
  );
}
