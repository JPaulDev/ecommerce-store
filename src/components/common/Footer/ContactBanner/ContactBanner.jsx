import Link from 'next/link';
import { MobilePhone, Headset } from '../../../icons';
import * as Styled from './styles';

export default function ContactBanner() {
  return (
    <Styled.Container>
      <Link href="/contact" passHref>
        <Styled.Link>
          <MobilePhone width={20} height={27} />
          01782-444455
        </Styled.Link>
      </Link>
      <Link href="/contact" passHref>
        <Styled.Link>
          <Headset width={25} height={25} />
          Live Chat
        </Styled.Link>
      </Link>
    </Styled.Container>
  );
}
