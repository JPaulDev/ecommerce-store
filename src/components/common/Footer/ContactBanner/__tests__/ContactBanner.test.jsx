import renderer from 'react-test-renderer';
import ContactBanner from '../index';

describe('ContactBanner component', () => {
  it('renders a banner with a contact number and a live chat link', () => {
    const tree = renderer.create(<ContactBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
