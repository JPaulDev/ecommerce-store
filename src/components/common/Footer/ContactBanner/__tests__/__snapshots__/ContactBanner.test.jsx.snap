// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactBanner component renders a banner with a contact number and a live chat link 1`] = `
.c0 {
  background-color: var(--color-sky-blue-2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 0;
}

.c0 :last-child {
  border: 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-right: 2px solid white;
  color: white;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Uni Sans',sans-serif;
  font-size: var(--font-size-20);
  font-weight: var(--font-weight-semibold);
  padding: 0 25px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  color: var(--color-grey-11);
}

<div
  className="c0"
>
  <a
    className="c1"
    href="/contact"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <svg
      fill="currentColor"
      height={27}
      viewBox="0 0 384 512"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 64C16 28.7 44.7 0 80 0H304c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H80c-35.3 0-64-28.7-64-64V64zM224 448c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zM304 64H80V384H304V64z"
      />
    </svg>
    01782-444455
  </a>
  <a
    className="c1"
    href="/contact"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <svg
      fill="currentColor"
      height={25}
      viewBox="0 0 512 512"
      width={25}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 48C141.1 48 48 141.1 48 256v40c0 13.3-10.7 24-24 24s-24-10.7-24-24V256C0 114.6 114.6 0 256 0S512 114.6 512 256V400.1c0 48.6-39.4 88-88.1 88L313.6 488c-8.3 14.3-23.8 24-41.6 24H240c-26.5 0-48-21.5-48-48s21.5-48 48-48h32c17.8 0 33.3 9.7 41.6 24l110.4 .1c22.1 0 40-17.9 40-40V256c0-114.9-93.1-208-208-208zM144 208h16c17.7 0 32 14.3 32 32V352c0 17.7-14.3 32-32 32H144c-35.3 0-64-28.7-64-64V272c0-35.3 28.7-64 64-64zm224 0c35.3 0 64 28.7 64 64v48c0 35.3-28.7 64-64 64H352c-17.7 0-32-14.3-32-32V240c0-17.7 14.3-32 32-32h16z"
      />
    </svg>
    Live Chat
  </a>
</div>
`;
